### PR TITLE
[WIP] Add(Modal): Add compact mode 

### PR DESCRIPTION
**Pull request by conglei**
to: @milesj @stefhatcher @hayes @alecklandgraf

cc: @kenchendesign  @elibrumbaugh

## Description

This PR is to add the compact mode support for `Modal` component. Tests are added accordingly.

## Motivation and Context

The main goal of this iteration is to reduce the amount of space in the modal header, so that users will read large content more easily. In addition, this iteration added the design for scrollable content, stepped modals and modal appearing transition.

## Testing

Unit test coverage

## Screenshots

Screen Shot 2019-04-08 at 2 36 58 PM

## Checklist

- My code follows the style guide of this project.
- I have updated or added documentation accordingly.
- I have read the CONTRIBUTING document.

### CODE REVIEW

**Comment by milesj** [comment]
@conglei I was under the impression that we're completely changing modals, not just adding a `compact` prop. Has this changed?

File: packages/core/test/components/Modal.test.tsx [hunk 1]
```diff
@@ -219,6 +219,25 @@ describe('<Modal />', () => {
     expect(titleWrapper.prop('children')).toBe('Title wave');
   });
 
+  it('render a large titel when prop large is true', () => {
```
Comment (stefhatcher) [inline packages/core/test/components/Modal.test.tsx]: `title`

Comment (stefhatcher) [reply packages/core/test/components/Modal.test.tsx]: did you want to add a test for `large` or `compact`? please add a test for `compact` to reflect the changes in the PR

**Comment by conglei** [comment]
Thanks @milesj for the quick review!  This is still WIP, will add other functions and tests. 

The reason I added `compact` is to make sure the change won't affect any existing apps which use `Model`.  I'm not against completely changing the modal if that is the case :) 

cc:  @kenchendesign

**Comment by milesj** [comment]
@conglei As long as the API is backwards compatible, feel free to change the full design.

File: packages/core/test/components/Modal.test.tsx (hunk 1 above)
Comment (conglei) [reply packages/core/test/components/Modal.test.tsx]: Will do ! Thanks @stefhatcher

**Comment by stefhatcher** [comment]
thanks for contributing! closing in favor of https://github.com/airbnb/lunar/pull/108

**Pull request closed without merging**